sys/cddl/contrib/opensolaris/uts/common/sys/dtrace.h

```diff
@@ -1299,19 +1299,19 @@ typedef struct dtrace_providerdesc {
 							/* provider query */
 #define	DTRACEIOC_PROBES	_IOWR('x',2,dtrace_probedesc_t)
 							/* probe query */
-#define	DTRACEIOC_BUFSNAP	_IOW('x',4,dtrace_bufdesc_t *)	
+#define	DTRACEIOC_BUFSNAP	_IOW('x',4,dtrace_bufdesc_t *__kerncap)
 							/* snapshot buffer */
 #define	DTRACEIOC_PROBEMATCH	_IOWR('x',5,dtrace_probedesc_t)
 							/* match probes */
 typedef struct {
-	void	*dof;		/* DOF userland address written to driver. */
+	void * __kerncap dof;		/* DOF userland address written to driver. */
```

Comment:
In this way we break compatibility with a non-capability userspace.. which is the best way to keep it?

Reply:
This breaks capability for freebsd64 userspaces.  You'd need to add a new type with this members declared as a `uint64_t`, use _IOC_NEWTYPE to declare relevant `DTRACEIOC_*64` macros, and add them where the non-`64` variants are used.

Reply:
Done it!